frontend/src/pages/editRole/EditRole.tsx

```diff
@@ -1,75 +1,132 @@
+import { Row, Col, Form, Input, Button, Spin } from "antd";
 import UserList from "./components/UserList"
 import { ApiRoutes, GET_Responses, UserRole } from "../../@types/requests.d";
 import apiCall from "../../util/apiFetch";
+import { useTranslation } from "react-i18next";
+import { UsersListItem } from "./components/UserList";
 
 export type UsersType = GET_Responses[ApiRoutes.USERS]
 
 const ProfileContent = () => {
-    const [users, setUsers] = useState<UsersType[] | null>(null);
+    const [users, setUsers] = useState<UsersType | null>(null);
+    const [searched, setSearched] = useState(false);
+    const [form] = Form.useForm();
+    const { t } = useTranslation();
 
-    function updateRole(user: UsersType, role: UserRole) {
-      //TODO: PUT of PATCH call
-      console.log("User: ", user);
-      console.log("Role: ", role);
-      if(!users) return;
-      const updatedUsers = users.map((u) => {
-        if (u.userId === user.userId) {
-         return { ...u, role: role };
-        }
-        return u;
-      });
-      setUsers(updatedUsers);
+    function updateRole(user: UsersListItem, role: UserRole) {
+      //here user is of type User (not UsersListItem), but it seems to work because the needed properties are named the same
+      console.log(user)
+      apiCall.patch(ApiRoutes.USER, {role: role}, {id: user.id}).then((res) => {
+        console.log(res.data);
+        //replace this user in the userlist with the updated one from res.data
+        const updatedUsers = users?.map((u) => {
+          if (u.id === user.id) {
+            return { ...u, role: res.data.role };
+          }
+          return u;
+        });
+        setUsers(updatedUsers?updatedUsers:null);
+      })
     }
 
-    useEffect(() => {
-      //TODO: moet met GET call
-        /*apiCall.get(ApiRoutes.USERS).then((res) => {
-          console.log(res.data)
-          setUsers(res.data)
-        })*/
-      setUsers([
-        {
-          userId: 1,
-          name: "Alice Kornelis",
-          role: "student",
-          email: "test@test.test",
-          url: "test"
-        },
-        {
-          userId: 2,
-          name: "Bob Kornelis",
-          role: "teacher",
-          email: "test@test.test",
-          url: "test"
-        },
-        {
-          userId: 3,
-          name: "Charlie Kornelis",
-          role: "admin",
-          email: "test@test.test",
-          url: "test"
-        }
-      ]);
-      }, []);
+    const onSearch = (values: any) => {
+      setSearched(true);
+      setUsers(null);
+      //search operation here
+      const params = Object.entries(values)
+        .filter(([key, value]) => value !== undefined)
+        .reduce((obj, [key, value]) => ({ ...obj, [key]: value }), {});
+      const queryString = Object.entries(params)
+        .map(([key, value]) => `${key}=${value}`)
+        .join('&');
+      console.log(queryString);
+      apiCall.get(ApiRoutes.USERS,{params:queryString}).then((res) => {

```

Comment:
Om query strings toe te voegen aan je url kan je new URL gebruiken.
Als je dit doet zal ts ambetant doen, maar je kan het dan casten naar `ApiRoutes.USERS`. Je zal wel nog de "?:params" weg moeten halen in ApiRoutes enum.
Vb:


```ts
      const params = new URLSearchParams();
      params.append('email', 'value1');
      params.append('name', 'value2');
   
      apiCall.get(ApiRoutes.USERS+"?" + params.toString() as ApiRoutes.USERS).then((res) => {
        console.log(res.data)
        setUsers(res.data);
      })


```